### PR TITLE
Provide user_to_session and user_to_token to complement Sihl_user.Web

### LINKED ISSUE #473: Add Web.user_to_session and Web.user_to_token

### PR DESCRIPTION
**Pull request by jerben**
Fix #473

### CODE REVIEW

**Comment by jerben** [comment]
### Todo 
- [ ] write tests
- [ ] document usage `index.mld`

But can already be reviewed.

File: sihl/src/contract_user.ml [hunk 1]
```diff
@@ -34,10 +34,29 @@ module type Sig = sig
...
 
-    (** [user_from_session ?cookie_key ?secret ?key ?secret request] returns the
+    (** [user_to_token ?key create_token user response] returns a response with
+        a [Bearer] token set to the [Authorization] header that has the [user]
+        associated to it. This is typically part of the login workflow. Any
+        previously set [Authorization] header is overwritten. This is typically
+        part of the login workflow.
+
+        [create_token] is a function that associates data to a string.
+        [sihl-token] can be used here. *)
```
Comment (aronerben) [inline sihl/src/contract_user.ml:47]: Do you want to point to a specific function from this package?

File: sihl/src/contract_user.ml [hunk 2]
```diff
@@ -34,10 +34,29 @@ module type Sig = sig
...
+    val user_to_token
+      :  ?key:string
+      -> ((string * string) list -> string Lwt.t)
+      -> t
+      -> Opium.Response.t
+      -> Opium.Response.t Lwt.t
+
+    (** [user_from_session ?cookie_key ?secret ?key ?secret request] returns a
+        response with a [Bearer] token set to the [Authorization] header that
```
Comment (aronerben) [inline sihl/src/contract_user.ml:56]: Is this correct? This function seems to return a `t` which is a user here

File: sihl/src/contract_user.ml [hunk 3]
```diff
@@ -34,10 +34,29 @@ module type Sig = sig
     val user_from_token
       :  ?key:string
       -> (string -> k:string -> string option Lwt.t)
-      -> Rock.Request.t
+      -> Opium.Request.t
       -> t option Lwt.t
 
-    (** [user_from_session ?cookie_key ?secret ?key ?secret request] returns the
+    (** [user_to_token ?key create_token user response] returns a response with
```
Comment (aronerben) [inline sihl/src/contract_user.ml:40]: You don't explain `key` in the docs here

File: sihl/src/contract_user.ml [hunk 4]
```diff
@@ -52,8 +71,28 @@ module type Sig = sig
...
-      -> Rock.Request.t
+      -> Opium.Request.t
       -> t option Lwt.t
+
+    (** [user_to_session ?cookie_key ?secret ?key user response] returns a
+        response with a [user] associated to the session. Any previously set
+        session is overwritten. This is typically part of the login workflow.
+
+        [cookie_key] is the name/key of the session cookie. By default, the
+        value is [_session].
```
Comment (aronerben) [inline sihl/src/contract_user.ml:82]: I don't think you are setting that default in the function implementation

**Review by aronerben — CHANGES_REQUESTED** [review]

File: sihl/src/contract_user.ml (hunk 1 above)
Comment (jerben) [reply sihl/src/contract_user.ml:47]: No, `sihl` does not know `sihl-token`.

File: sihl/src/contract_user.ml (hunk 2 above)
Comment (jerben) [reply sihl/src/contract_user.ml]: Fixed.

File: sihl/src/contract_user.ml (hunk 4 above)
Comment (jerben) [reply sihl/src/contract_user.ml:80]: It is by `web_session`, we are forwarding the optional.

**Comment by jerben** [comment]
I just realized that `user_to_token` sets the authorization header which doesn't make any sense. I checked how passportjs does it and they are not concerned with the creation of the token at all, that is up to you. I will close this PR.

**Pull request closed without merging**